### PR TITLE
Added validation for find command options.

### LINKED ISSUE #349: find command options validation
- Allow skip only if there is sort
- Allow pagingState if there is no sort

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:

Validate find command options using validator. Closes #349 

**Which issue(s) this PR fixes**:
Fixes #349 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/api/configuration/ObjectMapperConfigurationTest.java [hunk 1]
```diff
@@ -177,37 +176,6 @@ public void happyPath() throws Exception {
     }
   }
 
-  @Nested
```
Comment (maheshrajamani) [inline src/test/java/io/stargate/sgv2/jsonapi/api/configuration/ObjectMapperConfigurationTest.java:180]: Moved it to FindCommandTest

**Review by tatu-at-datastax — APPROVED** [review]
LGTM

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/validation/FindOptionsValidation.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    if (options.skip() != null && value.sortClause() == null) {
+      context
+          .buildConstraintViolationWithTemplate("skip options should be used with sort clause")
+          .addPropertyNode("options.skip")
+          .addConstraintViolation();
+      return false;
+    }
+    if (options.skip() != null && options.skip() < 0) {
+      context
+          .buildConstraintViolationWithTemplate("skip should be grater than or equal to `0`")
```
Comment (vkarpov15) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/validation/FindOptionsValidation.java:24]: Typo "grater" -> "greater"

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/validation/FindOptionsValidation.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          .addPropertyNode("options.skip")
+          .addConstraintViolation();
+      return false;
+    }
+    if (options.limit() != null && options.limit() < 0) {
+      context
+          .buildConstraintViolationWithTemplate("limit should be grater than or equal to `0`")
```
Comment (vkarpov15) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/validation/FindOptionsValidation.java:31]: Typo "grater" -> "greater"

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/validation/FindOptionsValidation.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          .addPropertyNode("options.limit")
+          .addConstraintViolation();
+      return false;
+    }
+    if (options.pagingState() != null && value.sortClause() != null) {
```
Comment (vkarpov15) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/validation/FindOptionsValidation.java:36]: Just to confirm, you can still do `skip` and `limit` with sort right?

Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/validation/FindOptionsValidation.java:36]: Yes, skip and limit will work with sort

**Pull request merged**